src/test/fuzz/txrequest.cpp

```diff
@@ -0,0 +1,363 @@
+            do {
+                TXHASHES[txhash] = rng.rand256();
+            } while (*(TXHASHES[txhash].begin() + 31) != txhash || *(TXHASHES[txhash].begin()) != txhash);
+        }
+        for (int i = 0; i < 16; ++i) {
+            DELAYS[i] = std::chrono::microseconds{i};
+        }
+        for (int i = 16; i < 128; ++i) {
+            DELAYS[i] = DELAYS[i - 1] + std::chrono::microseconds{1 + rng.randbits(((i - 10) * 2) / 9)};
+        }
+        for (int i = 128; i < 256; ++i) {
+            DELAYS[i] = -DELAYS[255 - i];
+        }
+    }
+} g_initializer;
+
+/** Tester class for TxRequestTracker
+ *
+ * It includes a naive reimplementation of its behavior, for a limited set
+ * of MAX_TXHASHES distinct txids, and MAX_PEERS peer identifiers.
+ *
+ * All of the public member functions perform the same operation on
+ * an actual TxRequestTracker and on the state of the reimplementation.
+ * The output of GetRequestable is compared with the expected value
+ * as well.
+ *
+ * Check() calls the TxRequestTracker's sanity check, plus compares the
+ * output of the constant accessors (Size(), CountLoad(), CountTracked())
+ * with expected values.
+ */
+class Tester
+{
+    //! TxRequestTracker object being tested.
+    TxRequestTracker m_tracker;
+
+    //! States for txid/peer combinations in the naive data structure.
+    enum class State {
+        NOTHING, //!< Absence of this txid/peer combination
+
+        // Note that this implementation does not distinguish between BEST/NEW/OTHER variants of CANDIDATE.
+        CANDIDATE,
+        REQUESTED,
+        COMPLETED,
+    };
+
+    //! Sequence numbers, incremented whenever a new CANDIDATE is added.
+    uint64_t m_current_sequence{0};
+
+    //! List of future 'events' (all inserted reqtimes/exptimes). This is used to implement AdvanceToEvent.
+    std::priority_queue<std::chrono::microseconds, std::vector<std::chrono::microseconds>, std::greater<std::chrono::microseconds>> m_events;
+
+    //! Information about a txhash/peer combination.
+    struct Announcement
+    {
+        std::chrono::microseconds m_time;
+        uint64_t m_sequence;
+        State m_state{State::NOTHING};
+        bool m_preferred;
+        bool m_is_wtxid;
+        uint64_t m_priority; //!< Precomputed priority.
+    };
+
+    //! Information about all txhash/peer combination.
+    Announcement m_announcements[MAX_TXHASHES][MAX_PEERS];
+
+    //! The current time; can move forward and backward.
+    std::chrono::microseconds m_now{244466666};
+
+    //! Delete txhashes whose only announcements are COMPLETED.
+    void Cleanup(int txhash)
+    {
+        bool all_nothing = true;
+        for (int peer = 0; peer < MAX_PEERS; ++peer) {
+            const Announcement& ann = m_announcements[txhash][peer];
+            if (ann.m_state != State::NOTHING) {
+                if (ann.m_state != State::COMPLETED) return;
+                all_nothing = false;
+            }
+        }
+        if (all_nothing) return;
```

Comment:
I believe that `all_nothing == true` means that all `txhash`es have the `State::NOTHING`.
Furthermore, re-assigning `State::NOTHING` to them will be effectively noop, right?
So one could comment out this LOC. Or not?

Reply:
No worries. It's just avoiding the work in case there is nothing to do.